### PR TITLE
fixes fatal error while loading smart group with deleted custom field

### PR DESCRIPTION
**Pull request by kurund**

### CODE REVIEW

File: CRM/Core/BAO/CustomField.php [hunk 1]
```diff
@@ -562,6 +562,9 @@ public static function getKeyID($key, $all = FALSE) {
    * @throws CRM_Core_Exception
    */
   public static function getFieldObject($fieldID) {
+    if (empty(self::getField($fieldID))) {
+      return;
+    }
     $field = new CRM_Core_BAO_CustomField();
     $field->copyValues(self::getField($fieldID));
```
Comment (colemanw) [inline CRM/Core/BAO/CustomField.php:569]: I'd change this to a variable so the getField function doesn't have to be called twice.

**Comment by colemanw** [comment]
@kurund this function is marked `@deprecated`. Where is it being called from? And can that function be refactored to use `getField()` instead?

**Comment by kurund** [comment]
> @kurund this function is marked `@deprecated`. Where is it being called from? And can that function be refactored to use `getField()` instead?

It is called by https://github.com/civicrm/civicrm-core/blob/master/CRM/Contact/BAO/Query.php#L4768 in my version (CiviCRM v5.71.2)

**Comment by kurund** [comment]
May be we can close this.

**Comment by kurund** [comment]
this was fixed in https://github.com/civicrm/civicrm-core/pull/30104

**Pull request closed without merging**